frontend/src/components/modules/GameInterface/GameInterface.tsx

```diff
@@ -0,0 +1,258 @@
+  const [matchInfo, setMatchInfo] = useState<matchData>({
+    timerTime: 300,
+    players: [],
+    playerNames: [],
+    winner: undefined,
+    officials: ""
+  });
+
+  useEffect(() => {
+    if (matchId !== undefined) {
+      joinMatch(matchId);
+
+      return () => {
+        leaveMatch(matchId);
+      };
+    }
+  }, []);
+
+  const [open, setOpen] = useState(false);
+  const [selectedButton, setSelectedButton] = useState<string>("");
+  const [timer, setTimer] = useState<number>(matchInfo.timerTime);
+  const [isTimerRunning, setIsTimerRunning] = useState<boolean>(false);
+  const [playerColor, setPlayerColor] = useState<PlayerColor>("red");
+
+  const { matchId } = useParams();
+  const { userId } = useAuth();
+  const { matchInfo: matchInfoFromSocket } = useSocket();
+
+  useEffect(() => {
+    const getMatchData = async (): Promise<void> => {
+      try {
+        let matchPlayers: MatchPlayer[] = [];
+        const playersNames: string[] = [];
+        let matchWinner: string | undefined;
+        let officialId: string = "";
+        let time: number = 0;
+        if (matchInfoFromSocket !== undefined) {
+          matchPlayers = matchInfoFromSocket.players;
+          playersNames[0] = (
+            await api.user.details(matchInfoFromSocket.players[0].id)
+          ).firstName;
+          playersNames[1] = (
+            await api.user.details(matchInfoFromSocket.players[1].id)
+          ).firstName;
+          if (matchInfoFromSocket.winner !== undefined) {
+            matchWinner = (await api.user.details(matchInfoFromSocket.winner))
+              .firstName;
+          }
+          if (matchInfoFromSocket.officials !== undefined) {
+            officialId = matchInfoFromSocket.officials;
+          }
+          time = 300 - Math.round(matchInfoFromSocket.elapsedTime / 1000);
+        } else if (matchId !== undefined) {
+          const matchFromApi: Match = await api.match.info(matchId);
+          if (matchFromApi !== undefined) {
+            matchPlayers = matchFromApi.players;
+            playersNames[0] = (
+              await api.user.details(matchFromApi.players[0].id)
+            ).firstName;
+            playersNames[1] = (
+              await api.user.details(matchFromApi.players[1].id)
+            ).firstName;
+            if (matchFromApi.winner !== undefined) {
+              matchWinner = (await api.user.details(matchFromApi.winner))
+                .firstName;
+            }
+            if (matchFromApi.officials !== undefined) {
+              officialId = matchFromApi.officials;
+            }
+            time = 300 - Math.round(matchFromApi.elapsedTime / 1000);
+          }
+        }
+        setMatchInfo({
+          timerTime: time,
+          players: matchPlayers,
+          playerNames: playersNames,
+          winner: matchWinner,
+          officials: officialId
+        });
+      } catch (error) {
+        console.error("Data couldn't be fetched", error);
```

Comment:
Use the _useToast_ hook and pass the error object to it. We shouldn't use any console logs outside of development.